pkg/controller/migration/convert/network.go

```diff
@@ -16,34 +16,50 @@ const (
 )
 
 func handleNetwork(c *components, install *Installation) error {
-	// TODO: be more elegant about creation of this spec field if it already exists
-	if install.Spec.CalicoNetwork == nil {
-		install.Spec.CalicoNetwork = &operatorv1.CalicoNetworkSpec{}
+
+	// FELIX_DEFAULTENDPOINTTOHOSTACTION
+	defaultWepAction, err := c.node.getEnv(ctx, c.client, containerCalicoNode, "FELIX_DEFAULTENDPOINTTOHOSTACTION")
+	if err != nil {
+		return err
+	}
+	if defaultWepAction != nil && strings.ToLower(*defaultWepAction) != "accept" {
+		return ErrIncompatibleCluster{
+			fmt.Sprintf("unexpected FELIX_DEFAULTENDPOINTTOHOSTACTION: '%s'. Only 'accept' is supported.", *defaultWepAction),
+		}
 	}
 
+	return nil
+}
+
+func handleCalicoCNI(c *components, install *Installation) error {
+	plugin, err := getCNIPlugin(c)
+	if err != nil {
+		return err
+	}
+	if plugin != operatorv1.PluginCalico {
+		return nil
+	}
+
+	context := fmt.Sprintf("detected %s CNI plugin", plugin)
+
 	// CALICO_NETWORKING_BACKEND
 	netBackend, err := c.node.getEnv(ctx, c.client, containerCalicoNode, "CALICO_NETWORKING_BACKEND")
 	if err != nil {
 		return err
 	}
+
 	if netBackend != nil && *netBackend != "" && *netBackend != "bird" {
-		return ErrIncompatibleCluster{"only CALICO_NETWORKING_BACKEND=bird is supported at this time"}
+		return ErrIncompatibleCluster{fmt.Sprintf("%s, only CALICO_NETWORKING_BACKEND=bird is supported at this time", context)}
 	}
 
 	// Calico CNI
 	if c.calicoCNIConfig == nil {
-		return fmt.Errorf("no 'calico' cni conf found in CNI_NETWORK_CONFIG on install-cni")
+		return ErrIncompatibleCluster{fmt.Sprintf("%s, required cni config was not found ", context)}
 	}
 
-	// FELIX_DEFAULTENDPOINTTOHOSTACTION
-	defaultWepAction, err := c.node.getEnv(ctx, c.client, containerCalicoNode, "FELIX_DEFAULTENDPOINTTOHOSTACTION")
-	if err != nil {
-		return err
-	}
-	if defaultWepAction != nil && strings.ToLower(*defaultWepAction) != "accept" {
-		return ErrIncompatibleCluster{
-			fmt.Sprintf("unexpected FELIX_DEFAULTENDPOINTTOHOSTACTION: '%s'. Only 'accept' is supported.", *defaultWepAction),
-		}
+	// TODO: be more elegant about creation of this spec field if it already exists
```

Comment:
Not sure I follow - what more do you think we could do to resolve this TODO? This seems fine to me.

Reply:
I think this was a note that I just copied. But I agree I'm not sure what else would be done.